SampleOfferWithoutReward.sol

```diff
@@ -90,8 +124,9 @@ contract SampleOfferWithoutReward {
             isContractValid = false;
     }
 
+    // TODO: This function is quite inconvenient. See withdraw() alternative.
     function getDailyPayment() {
-        if (msg.sender != contractor)
```

Comment:
This is needed here mostly for semantics.

Indeed the money will end up with the contractor no matter who ends up calling this function, but the intention of this contract is that only the contractor should call this. Thus we have checks for anything that should not happen, like someone else calling the function for the contractor to withdraw money from the contract.